core/corehttp/gateway_handler.go

```diff
@@ -227,92 +229,102 @@ func (i *gatewayHandler) getOrHeadHandler(ctx context.Context, w http.ResponseWr
 		return
 	}
 
-	links, err := i.api.Unixfs().Ls(ctx, resolvedPath)
+	nd, err := i.api.ResolveNode(ctx, resolvedPath)
```

Comment:
Now that we get the whole node here, we can meld this call and `i.api.Unixfs().Cat(ctx, resolvedPath)` further above into one